docs/synapse/userguides/storm_adv_functions.rstorm

```diff
@@ -0,0 +1,777 @@
+  error.)
+
+Functions can be declared and invoked on their own, but are most often used when authoring more extensive
+Storm code to implement a set of related functionality, such as a Rapid Power-Up. A set of functions, each
+encapsulating Storm logic to perform a specific task, can work together to implement more complex
+capabilities. Given this architecture, it is common for functions to invoke other functions as part of
+their code, or to take the output of another function as an input parameter to perform another operation,
+as seen in some of the examples below.
+
+See the :ref:`dev_rapid_power_ups` section of the :ref:`devguide` for a more in-depth discussion of
+how to integrate multiple Storm components into a larger package.
+
+.. _storm-func-callable:
+
+Callable Functions
+++++++++++++++++++
+
+**Callable** functions are "regular" functions, similar to those in other programming languages.  A callable
+function is invoked (called) and returns a value using a ``return()`` statement. A ``return()`` statement
+**must** be present for a callable function to execute properly even if the function does not return a
+specific value.
+
+Callable functions are **executed in their entirety** before returning. They return **exactly** one value.
+
+.. TIP::
+  
+  Callable functions may contain multiple ``return()`` statements, based on the function's logic. The
+  **first** ``return()`` encountered during the function's execution will cause the function to stop
+  execution and return. If you are performing multiple actions within the function and want to ensure they
+  all complete before the function returns, place the ``return()`` in a `fini block`_ so it executes
+  once at the end of the function's pipeline.
+
+Use Cases
+*********
+
+Callable functions can be used to:
+
+- Check a condition and return a status (e.g., ``return((0))`` vs. ``return((1))``).
+- Return a value (such as a count).
+- Return a single node.
+- Perform isolated operations on a node in the pipeline.
+- Retrieve data from an external API.
+
+Pseudocode
+**********
+
+::
+  
+  function callable() {
+  
+      <do stuff>
+      return()
+  }
+
+Examples
+********
+
+**Return a node**
+
+A callable function can take input and attempt to create (or lift) a node.
+
+::
+  
+  //Takes a value expected to be an IPv4 or IPv6 as input
+  function makeIP(ip) {
+      
+      //Attempt to create (or lift) an IPv4 from the input
+      //Return the IPv4 and exit if successful
+      [ inet:ipv4 ?= $ip ]
+      return($node)
+      
+      //Otherwise, atempt to create (or lift) an IPv6 from the input
+      //Return the IPv6 and exit if successful
+      [ inet:ipv6 ?= $ip ]
+      return($node)
+      
+      //If the input is not a valid IPv4 or IPv6, the function
+      // will execute but will not return a node.
+  }
+  
+  //Invoke the function with the specified input and
```

Comment:
space between `// Invoke`

Reply:
No space at the start of the comment, space in the second / continuing line of the comment (consistent with rest of docs - or else I can change all of them):

```
  //Invoke the function with the specified input and
  // yield the result (if any) into the pipeline

```